src/mvesuvio/analysis_reduction.py

```diff
@@ -1,1114 +1,1053 @@
-    mN = 1.008  # a.m.u.
-    Ef = 4906.0  # meV
-    en_to_vel = 4.3737 * 1.0e-4
-    vf = np.sqrt(Ef) * en_to_vel  # m/us
-    hbar = 2.0445
-    return mN, Ef, en_to_vel, vf, hbar
-
-
-def pseudoVoigt(x, sigma, gamma, IC):
-    """Convolution between Gaussian with std sigma and Lorentzian with HWHM gamma"""
-    fg, fl = 2.0 * sigma * np.sqrt(2.0 * np.log(2.0)), 2.0 * gamma
-    f = 0.5346 * fl + np.sqrt(0.2166 * fl**2 + fg**2)
-    eta = 1.36603 * fl / f - 0.47719 * (fl / f) ** 2 + 0.11116 * (fl / f) ** 3
-    sigma_v, gamma_v = f / (2.0 * np.sqrt(2.0 * np.log(2.0))), f / 2.0
-    pseudo_voigt = eta * lorentizian(x, gamma_v) + (1.0 - eta) * gaussian(x, sigma_v)
-
-    norm = (
-        np.abs(np.trapz(pseudo_voigt, x, axis=1))[:, np.newaxis] if IC.normVoigt else 1
-    )
-    return pseudo_voigt / norm
+        lorentzianResWidth = np.sqrt(dWdE1_lor + dQdE1_lor) * dE1_lorz  # in A-1
+        return lorentzianResWidth
 
 
-def gaussian(x, sigma):
-    """Gaussian function centered at zero"""
-    gaussian = np.exp(-(x**2) / 2 / sigma**2)
-    gaussian /= np.sqrt(2.0 * np.pi) * sigma
-    return gaussian
+    def pseudoVoigt(self, x, sigma, gamma):
+        """Convolution between Gaussian with std sigma and Lorentzian with HWHM gamma"""
+        fg, fl = 2.0 * sigma * np.sqrt(2.0 * np.log(2.0)), 2.0 * gamma
+        f = 0.5346 * fl + np.sqrt(0.2166 * fl**2 + fg**2)
+        eta = 1.36603 * fl / f - 0.47719 * (fl / f) ** 2 + 0.11116 * (fl / f) ** 3
+        sigma_v, gamma_v = f / (2.0 * np.sqrt(2.0 * np.log(2.0))), f / 2.0
+        pseudo_voigt = eta * lorentzian(x, gamma_v) + (1.0 - eta) * gaussian(x, sigma_v)
 
+        norm = (
+            np.abs(np.trapz(pseudo_voigt, x, axis=1))[:, np.newaxis] if self._run_norm_voigt else 1
+        )
+        return pseudo_voigt / norm
+
+
+    # When interface is updated, uncomment to change the way
+    # constraints are handled:
+
+    # def _get_parsed_constraints(self):
+    #
+    #     parsed_constraints = []
+    #
+    #     for constraint in  self._constraints:
+    #         constraint['fun'] = self._get_parsed_constraint_function(constraint['fun']) 
+    #
+    #         parsed_constraints.append(constraint)
+    #
+    #     return parsed_constraints
+    #
+    #
+    # def _get_parsed_constraint_function(self, function_string: str):
+    #
+    #     profile_order = [key for key in self._profiles.keys()]
+    #     attribute_order = ['intensity', 'width', 'center']
+    #
+    #     words = function_string.split(' ')
+    #     for i, word in enumerate(words):
+    #         if '.' in word:
+    #
+    #             try:    # Skip floats 
+    #                 float(word) 
+    #             except ValueError: 
+    #                 continue
+    #
+    #             profile, attribute = word
+    #             words[i] = f"pars[{profile_order.index(profile) + attribute_order.index(attribute)}]" 
+    #
+    #     return eval(f"lambda pars: {' '.join(words)}")
```

Comment:
Why not uncomment it now and just not call these functions?

Reply:
If it's not tested, better to delete it and recover from version control history later!